src/Storage/FileStorage.php

```diff
@@ -49,13 +44,15 @@ public function getData(): array
 
     public function flush(): void
     {
-        $varDumper = VarDumper::create($this->getData());
-        $jsonData = $varDumper->asJson();
-        $this->filesystem->write($this->path . '/' . $this->debugId . '.data.json', $jsonData);
-
-        $jsonObjects = $varDumper->asJsonObjectsMap();
-        $this->filesystem->write($this->path . '/' . $this->debugId . '.obj.json', $jsonObjects);
-
-        $this->collectors = [];
+        try {
+            $varDumper = VarDumper::create($this->getData());
+            $jsonData = $varDumper->asJson();
+            $this->filesystem->write($this->path . '/' . $this->idGenerator->getId() . '.data.json', $jsonData);
+
+            $jsonObjects = $varDumper->asJsonObjectsMap();
+            $this->filesystem->write($this->path . '/' . $this->idGenerator->getId() . '.obj.json', $jsonObjects);
+        } finally {
+            $this->collectors = [];
```

Comment:
I think now is no needs to empty the collectors, because you do `reset()` to all collectors

Reply:
No, they will be added on the next debugger start.

Reply:
Why?